### ISSUE TITLE
A single shortcut to proofread only the newly typed text

### ISSUE THREAD

**Issue by wgnmedia** [micz/ThunderAI#196]
I would love to see an option for a direct shortcut to send the newly typed text to ChatGPT without needing to activate the prompts menu or manually select the text. This could be implemented by automatically extracting the text above the "On DATE, NAME wrote" line in the case of a reply, or using the entire text if it's a new message.  

I believe this is the most common use case for most users, so having a one-click, more efficient way to achieve this would greatly enhance the experience.  

Thank you for your wonderful add-on.

**Comment by micz** [comment]
I'll check if this is possibile.
It should get the text before `<div class="moz-cite-prefix">...</div>` element.

Thank you!

**Comment by wgnmedia** [comment]
Thinking about it, all it needs are:

1. An option for a shortcut to a specific prompt
2. An option for auto submit the text above <div class="moz-cite-prefix">...</div>

I also assume the same element mentioned above works the same for forward as it works for reply.

Thank you.

**Comment by micz** [comment]
It's not easy to find free shortcuts, so I don't think adding a new one for a single prompt is a viable solution.
You can use the shortcut to open the menu and press a number to submit a prompt, without pressing the enter key.

I was thinking to add a new data placeholder to get the text as you asked, so it could be used also on a custom prompt.
It should work also when forwarding.

**Comment by micz** [comment]
To be released in version 3.1.0

**Comment by micz** [comment]
I released version [3.1.0pre1](https://github.com/micz/ThunderAI/releases) that has also this feature.
Check the release notes.

**Comment by wgnmedia** [comment]
Thank you. I'll give it a try in the next couple of weeks.

**Comment by micz** [comment]
Are you using the chatgpt web interface?

**Comment by wgnmedia** [comment]
No. I'm using ChatGPT API.

**Comment by micz** [comment]
Ok, now I understand what you meant. It can not substitute what it's not selected.
I'll think about it and report back here.
Thank you for the feedback.

**Comment by micz** [comment]
You can try version [3.1.0pre2](https://github.com/micz/ThunderAI/releases) that has #229.

**Comment by wgnmedia** [comment]
I installed 3.1.0pre3 and it works perfectly.

Thank you!

**Comment by micz** [comment]
May you tell me which Thunderbird version are you using in which Operative System?
Thank you.

**Comment by wgnmedia** [comment]
128.6.0esr (64-bit) on Windows 10.

**Comment by micz** [comment]
If you use a default prompt that requires to manually select the text, like "rewrite formal", does it work?

**Comment by wgnmedia** [comment]
No. only when I manually copy the text and CTRL + SHIFT + V it works. I believe that this should be the behavior for all prompts.

**Comment by micz** [comment]
For me it works every time on different PCs on any prompt.
Do you have nay particular addon installed for the message composition?

![Image](https://github.com/user-attachments/assets/0a24f5a3-dd54-4931-bee3-3e3ca0f85b44)
In the composition window what is set here?

Some users are asking the AI to format the text with bullet points, bold or italic text, so ThunderAI is keeping that formatting.
In my setup it is always retaining the formatting and line-breaking.

**Comment by micz** [comment]
Are you composing the emails in text only?

**Comment by micz** [comment]
> Are you composing the emails in text only?

I tried also composing in plain text only, but it keeps the line breaks.
Are you using any other add-ons that might interfere with the text in the message composition window?

**Comment by micz** [comment]
Released in version [3.1.0](https://github.com/micz/ThunderAI/releases).

**Comment by wgnmedia** [comment]
I don't have any add-ons that interfere, as far as I know. I tried both Body text and Paragraph, but it doesn't retain the line breaks.

Anyway, thank you for your help.

**Comment by micz** [comment]
I'm sorry, I really don't know what could be causing this behavior.
If I post a version with more logs, could you try it?

**Comment by wgnmedia** [comment]
I'll try first to play with the add-ons and other settings before you make a special release. 

Thank you.
